### PR TITLE
charm push: fix adding extra-info and resources

### PR DESCRIPTION
**Pull request by frankban**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/63/
Test PASSed.

**Comment by frankban** [comment]
This fixes #43

**Comment by mhilton** [comment]
:+1:

File: cmd/charm/charmcmd/push_test.go [hunk 1]
```diff
@@ -351,34 +351,26 @@ func (s *pushSuite) TestMapLogEntriesToVcsRevisions(c *gc.C) {
...
-		foo, err := os.Create(tempDir + "/foo")
-		c.Assert(err, gc.IsNil)
-		defer foo.Close()
-		_, err = foo.WriteString("bar")
-		c.Assert(err, gc.IsNil)
-	}
+
+	foo, err := os.Create(tempDir + "/foo")
+	c.Assert(err, gc.IsNil)
+	defer foo.Close()
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/push_test.go]: I think the old code was perhaps correct to close foo before
doing the git stuff, as under some platforms (e.g. Windows)
you won't be able to read the file while it's still open.

Why not just use ioutil.WriteFile instead?

File: cmd/charm/charmcmd/push_test.go [hunk 2]
```diff
@@ -398,25 +390,17 @@ func hg(c *gc.C, tempDir string, arg ...string) {
...
 	hg(c, tempDir, "init")
-	{
-		foo, err := os.Create(tempDir + "/foo")
-		c.Assert(err, gc.IsNil)
-		defer foo.Close()
-		_, err = foo.WriteString("bar")
-		c.Assert(err, gc.IsNil)
-	}
+
+	foo, err := os.Create(tempDir + "/foo")
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/push_test.go]: ditto

**Comment by rogpeppe** [comment]
LGTM with a couple of suggestions.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/64/
Test PASSed.

File: cmd/charm/charmcmd/push_test.go (hunk 1 above)
Comment (frankban) [inline cmd/charm/charmcmd/push_test.go]: Done.

File: cmd/charm/charmcmd/push_test.go (hunk 2 above)
Comment (frankban) [inline cmd/charm/charmcmd/push_test.go]: Done.

**Comment by rogpeppe** [comment]
LGTM thanks!

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**